services/ssooidc/src/main/java/software/amazon/awssdk/services/ssooidc/SsoOidcProfileTokenProviderFactory.java

```diff
@@ -42,6 +45,58 @@ public SdkTokenProvider create(ProfileFile profileFile, Profile profile) {
         return new SsooidcProfileTokenProvider(profileFile, profile);
     }
 
+    public SdkTokenProvider create(Supplier<ProfileFile> profileFile, String profileName) {
+        return new SsooidcSuppliedProfileTokenProvider(profileFile, profileName);
+    }
+
+    private static final class SsooidcSuppliedProfileTokenProvider implements SdkTokenProvider, SdkAutoCloseable {
```

Comment:
Nit: Can we capitalize the `o` in `SsooidcSuppliedProfileTokenProvider` to get `SsoOidcSuppliedProfileTokenProvider`? And preferably change the existing `SsooidcProfileTokenProvider` too

Reply:
Good catch. Updated both classes.